src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,151 @@
+    {
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="configure">A callback to configure the connection.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
+        public static IEndpointConventionBuilder MapConnections(this IEndpointRouteBuilder builder, string pattern, Action<IConnectionBuilder> configure) =>
+            builder.MapConnections(pattern, new HttpConnectionDispatcherOptions(), configure);
+
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <typeparam name="TConnectionHandler">The <see cref="ConnectionHandler"/> type.</typeparam>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
+        public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder builder, string pattern) where TConnectionHandler : ConnectionHandler
+        {
+            return builder.MapConnectionHandler<TConnectionHandler>(pattern, configureOptions: null);
+        }
+
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <typeparam name="TConnectionHandler">The <see cref="ConnectionHandler"/> type.</typeparam>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="configureOptions">A callback to configure dispatcher options.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
+        public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder builder, string pattern, Action<HttpConnectionDispatcherOptions> configureOptions) where TConnectionHandler : ConnectionHandler
+        {
+            var options = new HttpConnectionDispatcherOptions();
+            // REVIEW: WE should consider removing this and instead just relying on the
+            // AuthorizationMiddleware
+            var attributes = typeof(TConnectionHandler).GetCustomAttributes(inherit: true);
+            foreach (var attribute in attributes.OfType<AuthorizeAttribute>())
+            {
+                options.AuthorizationData.Add(attribute);
+            }
```

Comment:
This kind of thing (doing reflection up front) is a nice simple thing that makes me happy.